metis-indexing/src/main/java/eu/europeana/indexing/ConnectionProvider.java

```diff
@@ -0,0 +1,46 @@
+package eu.europeana.indexing;
+
+import java.io.Closeable;
+import org.apache.solr.client.solrj.SolrClient;
+import eu.europeana.corelib.mongo.server.EdmMongoServer;
+import eu.europeana.indexing.mongo.FullBeanDao;
+
+/**
+ * <p>
+ * This class is maintainable for providing an instance of {@link Indexer} with the required
+ * connections to persistence storage. It is responsible for maintaining the connections and
+ * providing access to a Publisher object (instance of {@link FullBeanPublisher}) for publishing
+ * Full Beans to be accessed by external agents.
+ * </p>
+ * <p>
+ * Please note that this class is {@link Closeable} and must be closed to release it's resources.
+ * </p>
+ * 
+ * @author jochen
+ *
+ */
+abstract class ConnectionProvider implements Closeable {
```

Comment:
Let's start using this rule for naming abstract classes?
http://grepcode.com/file/repo1.maven.org/maven2/org.codehaus.sonar-plugins.java/java-checks/1.3/org/sonar/l10n/squidjava/rules/squid/S00118.html

Reply:
Are we strict about it? I changed the name to AbstractConnectionProvider, but this name would not be valid according to the link you provided because two words follow after Abstract instead of one.